routing/http/client/client.go

```diff
@@ -79,12 +82,18 @@ func WithIdentity(identity crypto.PrivKey) Option {
 	}
 }
 
+// WithHTTPClient sets a custom HTTP Client to be used with [Client]. If you use
+// this option, [WithUserAgent] will no longer work, as it is made to work with
+// the default HTTP Client.
```

Comment:
Is this true? Seems like setting the user agent will work as long as the implementation `httpClient` is `*http.Client` and it has a `ResponseBodyLimitedTransport`.